lib/ActivityStreams.js

```diff
@@ -132,17 +142,19 @@ ActivityStreams.prototype = {
   },
 
   /**
-   * Broadcast places changes to pages
+   * Handles changes to places
    */
   _handlePlacesChanges(eventName, data) {
-    this.broadcast(am.actions.Response("RECEIVE_PLACES_CHANGES", data));
-  },
-
-  /**
-   * Broadcast bookmark changes to pages
-   */
-  _handleBookmarkChanges(eventName, data) {
-    this.broadcast(am.actions.Response("RECEIVE_BOOKMARKS_CHANGES", data));
+    /* note: this will execute for each of the 3 notifications that occur
+     * when adding a visit: frecency:-1, frecency: real frecency, title */
+    this._invalidatePlacesCache();
+    this._asyncBuildPlacesCache();
+
```

Comment:
We can probably put the number of updates down to 10 rebuilds if we detect a new link and build a state machine.